### PR TITLE
Add the macros with the hardcoded prefixes

### PR DESCRIPTION
**Pull request by singh264**
Add the macros with the hardcoded prefixes
of format O\^O [a-z A-Z]+(:)?.

Closes: https://github.com/eclipse/omr/issues/4550.
Author: Amarpreet Singh amarpreet1997@gmail.com.

### CODE REVIEW

**Comment by singh264** [comment]
The local test results of the code changes in the GitHub pull request is located at the following url: https://drive.google.com/file/d/1OuVzManwMle7SJqRELQP315tafN6YX3U/view?usp=sharing.

The local test results of the code changes in the GitHub pull request could be displayed with the file that is located at the following url: https://github.com/singh264/scripts/blob/main/display_test_openj9-openjdk-jdk8_results.sh.

**Comment by singh264** [comment]
@VermaSh

**Comment by VermaSh** [comment]
Sorry just saw this PR, will review the changes later this week

File: compiler/codegen/CodeGenRA.cpp [hunk 1]
```diff
@@ -490,7 +493,7 @@ OMR::CodeGenerator::allocateSpill(int32_t dataSize, bool containsCollectedRefere
          }
 
      if (
-         (spill && self()->comp()->getOption(TR_TraceRA) && !performTransformation(self()->comp(), "O^O SPILL TEMPS: Reuse spill temp %s\n", self()->getDebug()->getName(spill->getSymbolReference()))))
+         (spill && self()->comp()->getOption(TR_TraceRA) && !performTransformation(self()->comp(), "%sReuse spill temp %s\n", OPT_DETAILS_SPILL_TEMPS, self()->getDebug()->getName(spill->getSymbolReference()))))
```
Comment (VermaSh) [inline compiler/codegen/CodeGenRA.cpp:496]: We can use generic `OPT_DETAILS` since `SPILL TEMPS` doesn't add anything we can't already make out from the transformation message

File: compiler/codegen/CodeGenRA.cpp [hunk 2]
```diff
@@ -77,6 +77,9 @@
 #include "ras/Debug.hpp"
 
 #define OPT_DETAILS "O^O CODE GENERATION: "
+#define OPT_DETAILS_SPILL_TEMPS "O^O SPILL TEMPS: "
+#define OPT_DETAILS_HALF_SLOT_SPILLS "O^O HALF-SLOT SPILLS: "
+#define OPT_DETAILS_GLOBAL_REGISTER_ALLOCATION "O^O GLOBAL REGISTER ALLOCATION: "
```
Comment (VermaSh) [inline compiler/codegen/CodeGenRA.cpp:82]: Doesn't look like we use `OPT_DETAILS` anywhere so maybe just update to say `CODE GENERATION REGISTER ALLOCATION: `.

File: compiler/codegen/CodeGenRA.cpp [hunk 3]
```diff
@@ -535,7 +538,7 @@ OMR::CodeGenerator::allocateSpill(int32_t dataSize, bool containsCollectedRefere
       //
       if (  offset == NULL
          || spill->secondHalfIsOccupied()
-            || !performTransformation(self()->comp(), "O^O HALF-SLOT SPILLS: Use second half of %s\n", self()->getDebug()->getName(spill->getSymbolReference())))
+            || !performTransformation(self()->comp(), "%sUse second half of %s\n", OPT_DETAILS_HALF_SLOT_SPILLS, self()->getDebug()->getName(spill->getSymbolReference())))
```
Comment (VermaSh) [inline compiler/codegen/CodeGenRA.cpp:541]: We can probably get away with using generic `OPT_DETAILS` here as well

File: compiler/il/Aliases.cpp [hunk 4]
```diff
@@ -65,6 +65,8 @@
 #include "runtime/RuntimeAssumptions.hpp"
 #endif
 
+#define OPT_DETAILS "O^O REFINING ALIASES: "
```
Comment (VermaSh) [inline compiler/il/Aliases.cpp:68]: Don't think we need to be specific here. `IL ALIASES: ` should suffice.

File: compiler/il/OMRResolvedMethodSymbol.cpp [hunk 5]
```diff
@@ -70,6 +70,7 @@
 #include "runtime/Runtime.hpp"
 #include "infra/SimpleRegex.hpp"
 
+#define OPT_DETAILS "O^O OSR: "
```
Comment (VermaSh) [inline compiler/il/OMRResolvedMethodSymbol.cpp:73]: We can reword this to be more generic since the message has enough detail to indicate the transformation is related to OSR call. So maybe reword to `IL Resolved Method Symbol`.

File: compiler/optimizer/Inliner.cpp [hunk 6]
```diff
@@ -1683,7 +1684,7 @@ void TR_InlinerBase::rematerializeCallArguments(TR_TransformInlinedFunction & ti
             if (rematTree == argStoreTree)
                {
                TR::Node *duplicateStore = argStore->duplicateTree();
-               if (performTransformation(comp(), "O^O GUARDED CALL REMAT: Rematerialize [%p] as [%p]\n", argStore, duplicateStore))
+               if (performTransformation(comp(), "%sRematerialize [%p] as [%p]\n", OPT_DETAILS_GUARDED_CALL_REMAT, argStore, duplicateStore))
```
Comment (VermaSh) [inline compiler/optimizer/Inliner.cpp:1687]: We can reword the message to indicate that a guarded call is being rematerialized.

File: compiler/optimizer/Inliner.cpp [hunk 7]
```diff
@@ -1693,7 +1694,7 @@ void TR_InlinerBase::rematerializeCallArguments(TR_TransformInlinedFunction & ti
                {
                TR::Node *duplicateStore = TR::Node::createStore(argStore->getSymbolReference(), TR::Node::createLoad(argStore, rematTree->getNode()->getSymbolReference()));
                duplicateStore->setByteCodeInfo(argStore->getByteCodeInfo());
-               if (performTransformation(comp(), "O^O GUARDED CALL REMAT: Partial rematerialize of [%p] as [%p] - load of [%d]\n", argStore, duplicateStore, rematTree->getNode()->getSymbolReference()->getReferenceNumber()))
+               if (performTransformation(comp(), "%sPartial rematerialize of [%p] as [%p] - load of [%d]\n", OPT_DETAILS_GUARDED_CALL_REMAT, argStore, duplicateStore, rematTree->getNode()->getSymbolReference()->getReferenceNumber()))
```
Comment (VermaSh) [inline compiler/optimizer/Inliner.cpp:1697]: We can reword the message to indicate that a guarded call is being partially rematerialized.

File: compiler/optimizer/Inliner.cpp [hunk 8]
```diff
@@ -119,6 +119,7 @@ int32_t  *NumInlinedMethods   = NULL;
 int32_t  *InlinedSizes        = NULL;
 
 #define OPT_DETAILS "O^O INLINER: "
+#define OPT_DETAILS_GUARDED_CALL_REMAT "O^O GUARDED CALL REMAT: "
```
Comment (VermaSh) [inline compiler/optimizer/Inliner.cpp:122]: Don't think we need a specific macro for guarded calls

**Review by VermaSh — CHANGES_REQUESTED** [review]
Some minor suggestions. I still have to go through 14 files, will post another comment for those soon.

File: compiler/optimizer/Inliner.cpp (hunk 6 above)
Comment (singh264) [reply compiler/optimizer/Inliner.cpp:1717]: Rematerilzed is being conveyed by this message and if not I would like your recommendation on a good message.

File: compiler/optimizer/Inliner.cpp (hunk 7 above)
Comment (singh264) [reply compiler/optimizer/Inliner.cpp]: Rematerilzed is being conveyed by this message and if not I would like your recommendation on a good message.

File: compiler/optimizer/Inliner.cpp (hunk 8 above)
Comment (singh264) [reply compiler/optimizer/Inliner.cpp:121]: Calls for guarded code can be good to remove provided no tests fails from this change.

**Comment by singh264** [comment]
https://github.com/singh264/scripts/blob/main/display_test_openj9-openjdk-jdk8_results.sh is not good anymore, @VermaSh I would like you to review this PR as it achieves the same kind of improvement as in the OpenJ9 PR that is similar to this PR.
